syntax = "proto3";

package exarp.bridge;

option go_package = "github.com/davidl71/exarp-go/proto";

// ToolRequest represents a request to execute a Python tool via bridge
message ToolRequest {
  // Name of the tool to execute
  string tool_name = 1;
  
  // Tool arguments as JSON string (for flexibility during migration)
  // Future: Can be replaced with tool-specific protobuf messages
  string arguments_json = 2;
  
  // Project root directory path
  string project_root = 3;
  
  // Timeout in seconds (0 = use default)
  int32 timeout_seconds = 4;
  
  // Request ID for tracking/correlation
  string request_id = 5;
}

// ToolResponse represents the response from a Python tool execution
message ToolResponse {
  // Whether the tool execution was successful
  bool success = 1;
  
  // Result text/output from the tool
  string result = 2;
  
  // Error message if execution failed
  string error = 3;
  
  // Execution time in milliseconds
  int64 execution_time_ms = 4;
  
  // Request ID (echoed from request)
  string request_id = 5;
  
  // Exit code from the tool process
  int32 exit_code = 6;
}

// ResourceRequest represents a request to execute a Python resource handler
message ResourceRequest {
  // Resource URI to fetch
  string uri = 1;
  
  // Project root directory path
  string project_root = 2;
  
  // Request ID for tracking
  string request_id = 3;
}

// ResourceResponse represents the response from a Python resource handler
message ResourceResponse {
  // Whether the resource fetch was successful
  bool success = 1;
  
  // Resource content (binary data)
  bytes content = 2;
  
  // MIME type of the resource
  string mime_type = 3;
  
  // Error message if fetch failed
  string error = 4;
  
  // Request ID (echoed from request)
  string request_id = 5;
}
